coordinator/src/collaborative_revert.rs

```diff
@@ -25,257 +26,280 @@ use dlc_manager::subchannel::LnDlcChannelSigner;
-        false => {
-            sub_channel.fund_value_satoshis as i64
-                - (channel_details.inbound_capacity_msat / 1000) as i64
-                - settlement_amount as i64
-                - (dlc_channel_fee as f64 / 2.0) as i64
-                - channel_details.unspendable_punishment_reserve.unwrap_or(0) as i64
+            let reserve = fund_txo_sat / Decimal::ONE_HUNDRED;
+
+            // The minimum reserve is _at least_ 1_000 sats.
+            let reserve = reserve.max(Decimal::ONE_THOUSAND);
+
+            reserve.to_u64().expect("Reserve to fit into u64")
         }
-        true => {
-            sub_channel.fund_value_satoshis as i64
-                - (channel_details.inbound_capacity_msat / 1000) as i64
-                - channel_details.unspendable_punishment_reserve.unwrap_or(0) as i64
+    };
+
+    let is_channel_split = match channel_details_option {
+        Some(ChannelDetails {
+            funding_txo: Some(funding_txo),
+            original_funding_outpoint: Some(original_funding_txo),
+            ..
+        }) => {
+            // The channel _is_ split if the `original_funding_txo` and the `funding_txo` differ.
+            original_funding_txo != funding_txo
+        }
+        Some(ChannelDetails {
+            funding_txo: Some(_),
+            original_funding_outpoint: None,
+            ..
+        }) => {
+            // The channel is _not_ split if the `original_funding_txo` has not been set.
+            false
+        }
+        Some(ChannelDetails {
+            funding_txo: None, ..
+        }) => {
+            bail!("Cannot collaboratively revert channel without funding TXO");
+        }
+        None => {
+            let params = without_channel_details_params.context(
+                "Missing ChannelDetails. Need WithoutChannelDetailsParams \
+                 to decide if we are dealing with a split channel",
+            )?;
+
+            params.is_channel_split
         }
     };
-    let trader_amount = sub_channel.fund_value_satoshis - coordinator_amount as u64;
 
-    let fee = weight_to_fee(
-        COLLABORATIVE_REVERT_TX_WEIGHT,
-        revert_params.fee_rate_sats_vb,
-    )
-    .expect("To be able to calculate constant fee rate");
-
-    tracing::debug!(
-        coordinator_amount,
-        fund_value_satoshis = sub_channel.fund_value_satoshis,
-        inbound_capacity_msat = channel_details.inbound_capacity_msat,
-        settlement_amount,
-        dlc_channel_fee,
-        inbound_capacity_msat = channel_details.inbound_capacity_msat,
-        outbound_capacity_msat = channel_details.outbound_capacity_msat,
-        trader_margin = position.trader_margin,
-        coordinator_margin = position.coordinator_margin,
-        position_id = position.id,
-        "Collaborative revert temporary values"
-    );
+    tracing::debug!(%is_channel_split, "Calculating coordinator revert amount");
+
+    let coordinator_amount = if is_channel_split {
```

Comment:
🤔 Shouldn't this be `!is_channel_split` ?

Reply:
Good catch! I was meaning to swap the `if`-`else` blocks instead, but I forgot.

This does highlight the fact that our test for this feature is very bad D: